pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -211,9 +431,9 @@ func (r *MachineSyncReconciler) shouldMirrorCAPIMachineToMAPIMachine(ctx context
 	logger.WithName("shouldMirrorCAPIMachineToMAPIMachine").
 		Info("checking if CAPI machine should be mirrored", "machine", machine.GetName())
 
-	// Check if owner refs point to CAPI MachineSet
+	// Check if owner refs point to CAPI Machine
 	for _, ref := range machine.ObjectMeta.OwnerReferences {
-		if ref.Kind != machineSetKind || ref.APIVersion != capiv1beta1.GroupVersion.String() {
+		if ref.Kind != machineKind || ref.APIVersion != capiv1beta1.GroupVersion.String() {
```

Comment:
This is still not resolved. It's not just the function comment, but every occurrence of the word MachineSet in this function was changed to Machine. This changes what the code does.

Reply:
Oops yes, thanks @RadekManak ! Good catch!
We need to add tests for those things anyway, will work on it on separate PRs in concert with @theobarberbany

Reply:
Fixed now.